src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp

```diff
@@ -20,24 +20,28 @@ ConvolutionKernel_b_fs_yx_fsv16_1x1::ConvolutionKernel_b_fs_yx_fsv16_1x1() : Con
     }
 }
 
-ConvolutionKernel_b_fs_yx_fsv16_1x1::AutoTuneOption ConvolutionKernel_b_fs_yx_fsv16_1x1::GetAutoTuneOptions(const Params& params,
+ConvolutionKernel_b_fs_yx_fsv16_1x1::AutoTuneOption ConvolutionKernel_b_fs_yx_fsv16_1x1::GetAutoTuneOptions(const convolution_params& params,
                                                                                                             int /*autoTuneIndex*/) const {
-    const convolution_params& cp = static_cast<const convolution_params&>(params);
-
-    auto x = cp.outputs[0].X().v;
-    auto y = cp.outputs[0].Y().v;
-    auto f = cp.outputs[0].Feature().v;
-
-    if (x == 1 && y == 1) {
-        return { 1, EXE_MODE_DEFAULT };
-    } else if (x * f <= 256) {
-        if (x < 8 || x * f <= 128)
-            return { 2, EXE_MODE_DEFAULT };
-        else
+    if (!params.has_dynamic_outputs()) {
+        auto x = params.outputs[0].X().v;
+        auto y = params.outputs[0].Y().v;
+        auto f = params.outputs[0].Feature().v;
+
+        if (x == 1 && y == 1) {
+            return { 1, EXE_MODE_DEFAULT };
+        } else if (x * f <= 256) {
+            if (x < 8 || x * f <= 128)
+                return { 2, EXE_MODE_DEFAULT };
+            else
+                return { 4, EXE_MODE_DEFAULT };
+        } else if (x * f <= 1536) {
             return { 4, EXE_MODE_DEFAULT };
-    } else if (x * f <= 1536) {
-        return { 4, EXE_MODE_DEFAULT };
+        } else {
+            return { 8, EXE_MODE_DEFAULT };
+        }
     } else {
+        // In shape agnostic kernel, the output shape cannot be specified at build time,
+        // So we initialy set blockWidth to 8 which is the most commonly used. Update blockWidth after static shape comes.
```

Comment:
Now this kernel can change blockwidth even in the dynamic case by use of multi-kernels. So, I think we can remove these comments.

Reply:
In my opinion, it would be better to have additional explanations rather than removing. So I wrote it in a different way.